oem/firstvoices/android/app/src/main/java/com/firstvoices/keyboards/FVShared.java

```diff
@@ -124,53 +129,83 @@ public static FVShared getInstance() {
   }
 
   private FVRegionList loadRegionList() {
-        FVRegionList list = new FVRegionList();
-        try {
-            // At this point in initialization, fv_all.kmp hasn't been extracted, so
-            // we get all the keyboard info from keyboards.csv
-            InputStream inputStream = context.getAssets().open("keyboards.csv");
-            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
-
-            reader.readLine(); // skip header row
-            String line = reader.readLine();
-
-            while (line != null) {
-                while (line.contains(",,"))
-                    line = line.replace(",,", ", ,");
-
-                String[] values = line.split(",");
-                if (values != null && values.length > 0) {
-                    // Read in column info
-                    String kbId = values[1];
-                    String kbName = values[2];
-                    String regionName = values[3];
-                    String legacyId = values[4];
-                    String version = values[5];
-                    String lgId = values[6].toLowerCase(); // Normalize language ID
-                    String lgName = values[7];
-
-                    FVRegion region = list.findRegion(regionName);
-                    if(region == null) {
-                        region = new FVRegion(regionName);
-                        list.add(region);
-                    }
-
-                    FVKeyboard keyboard = new FVKeyboard(kbId, kbName, legacyId, version, lgId, lgName);
-
-                    region.keyboards.add(keyboard);
-                }
+      FVRegionList list = new FVRegionList();
+      File resourceRoot = new File(getResourceRoot());
+      PackageProcessor kmpProcessor = new PackageProcessor(resourceRoot);
+      JSONParser parser = new JSONParser();
+      File jsonFile = new File(getPackagesDir() + FVDefault_PackageID + File.separator + FVKeyboards_JSON);
+      if (!jsonFile.exists()) {
+          return list;
+      }
+      try {
+          // At this point in initialization, fv_all.kmp is now extracted, so
+          // populate keyboard info from keyboards.json
+          JSONArray keyboardsArray = parser.getJSONObjectFromFile(jsonFile, JSONArray.class);
+
+          if (keyboardsArray == null) {
+              Log.d("loadRegionList", "unable to load keyboards.json");
+              return list;
+          }
```

Comment:
Malformed data should at the very least be reporting to sentry, or killing the app so we know that there is something wrong.

Reply:
I've updated the following to use KMLog to sentry.

Are you wanting to throw fatal errors for all of them?